### PR TITLE
Java: duplicates in to-many expands

### PR DESCRIPTION
**Pull request by MattSchur**
Describe how to avoid duplicates when expanding many-to-many associations

### CODE REVIEW

**Review by agoerler — APPROVED** [review]

File: java/working-with-cql/query-api.md [hunk 1]
```diff
@@ -450,9 +450,15 @@ To expand all first level associations of an entity, use `expand()` on the entit
...
-::::
+::: tip
+**Don't use distinct together with expand**
+
+The `distinct` clause removes duplicate rows from the root entity and effectively aggregates rows. Expanding child entities from aggregated rows is not well-defined and can lead to issues that can be resolved by removing distinct.
+
+**Resolving duplicates in to-many expands**
+
+Duplicates in to-many expands can occur on associations that are mapped as many-to-many without using a [link entity](../../guides/domain-modeling#many-to-many-associations) and don't correctly define the source cardinality. This can be resolved by adding the cardinality in the CDS model: `Association [*,*] to Entity`.
+:::
```
Comment (renejeglinsky) [inline java/working-with-cql/query-api.md:461]: How about splitting it into two tips?
image

Comment (agoerler) [reply java/working-with-cql/query-api.md:461]: +1 for splitting

* the first ("Don't use distinct together with expand") is rather a _warning_ from my point of view
* the second ("Resolving duplicates in to-many expands") 
  * is not related to "Optimized Expand Execution"
  * is maybe a general modelling advise 
     -> https://pages.github.tools.sap/cap/docs/guides/domain-modeling#many-to-many-associations ?

Comment (renejeglinsky) [reply java/working-with-cql/query-api.md:461]: Fine by me. Then I'll make the changes

Comment (MattSchur) [reply java/working-with-cql/query-api.md]: > is maybe a general modelling advise

please keep this in the Java expand section

Comment (renejeglinsky) [reply java/working-with-cql/query-api.md]: As in my screenshot/proposal?

Comment (MattSchur) [reply java/working-with-cql/query-api.md]: -> https://github.com/cap-js/docs/pull/876

**Pull request merged**